Shopify/shopify-cli
# Improve Bugsnag integration

### WHY are these changes introduced?

To link the stacktrace on Bugsnag to this repo, as shown here:

![bugsnag](https://user-images.githubusercontent.com/14979109/155318194-0979073b-b916-4c9a-8913-4e65bae36cfb.png)

Documentation: https://docs.bugsnag.com/product/releases/#source-control-information

### WHAT is this pull request doing?

- Adds a new workflow with [Bugsnag Report Build](https://github.com/marketplace/actions/bugsnag-report-build) GitHub Action to link the current build. As a first step, the version is hardcoded, but if this works, we could try to get the latest release of the repo.
- Prevents the Bugsnag report from the code to raise an error when it fails.

### How to test your changes?

For the workflow:
- Merge and push to main
- Check if the new Bugsnag GitHub action succeeds
- Check in Bugsnag if the links appear

For the error:
- Add a `raise` [here](https://github.com/Shopify/shopify-cli/blob/32cad820adc1c4417dca2eef042dc2809b87fed4/lib/shopify_cli/command.rb#L118) and [here](https://github.com/Shopify/shopify-cli/blob/10ccb3a7b842cad15cc6f5edb5a553e61b78ab65/lib/shopify_cli/exception_reporter.rb#L30) 
- Run `shopify`
- It should not show a stacktrace about the Bugsnag

### Update checklist
